airbyte-integrations/connectors/source-jira/source_jira/source.py

```diff
@@ -70,12 +71,12 @@ def _validate_and_transform_config(self, config: Mapping[str, Any]):
 
     @staticmethod
     def get_authenticator(config: Mapping[str, Any]):
-        return BasicHttpAuthenticator(config["email"], config["api_token"])
+        return BasicHttpAuthenticator(config.get("email"), config["api_token"])
```

Comment:
Why this change? email is required in the spec. Same comment with `domain` on line 79

Reply:
Yes, it should, but this line is executed before the check, so we fail with:
```
{"type":"TRACE","trace":{"type":"ERROR","emitted_at":1731587941523,"error":{"message":"Error starting the sync. This could be due to an invalid configuration or catalog. Please contact Support for assistance. Error: 'domain'","stack_trace":"Traceback (most recent call last):\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/source_jira/run.py\", line 22, in _get_source\n    return SourceJira(\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/source_jira/source.py\", line 24, in __init__\n    super().__init__(catalog=catalog, config=config, state=state, **{\"path_to_yaml\": \"manifest.yaml\"})\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/.source-jira/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/yaml_declarative_source.py\", line 33, in __init__\n    super().__init__(\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/.source-jira/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/concurrent_declarative_source.py\", line 81, in __init__\n    self._concurrent_streams, self._synchronous_streams = self._group_streams(\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/.source-jira/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/concurrent_declarative_source.py\", line 181, in _group_streams\n    for declarative_stream in self.streams(config=config):\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/source_jira/source.py\", line 62, in streams\n    return streams + self.get_non_portable_streams(config=config)\n  File \"/home/anatolii/airbytehq/airbyte/airbyte-integrations/connectors/source-jira/source_jira/source.py\", line 79, in get_non_portable_streams\n    args = {\"authenticator\": authenticator, \"domain\": config[\"domain\"], \"projects\": config[\"projects\"]}\nKeyError: 'domain'\n"}}}
```
With the get the error is correct:
```
{"type":"CONNECTION_STATUS","connectionStatus":{"status":"FAILED","message":"Config validation error: 'domain' is a required property"}}
```

Reply:
Ok! So the reason that it was failing properly before is that `AirbyteEntrypoint.validate_connection` was performed after the source is instantiated. Now, the instantiation of the source is failing. This is kind of annoying because to call `validate_connection`, we need the spec and to have the spec we need to instantiate the source. `Source.spec` method should probably be a classmethod but right now it isn't. So we are at risk of having an invalid config being provided an failing with a trace message instead of a connection_status failed.